{{ cookiecutter.repo_name }}/CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+# Contributing
+
+We love contributions! We've compiled this documentation to help you understand our contributing guidelines. If you
+still have questions, please [contact us][email] and we'd be happy to help!
+
+- [Code of Conduct](#code-of-conduct)
+- [Getting started](#getting-started)
+- [Code conventions](#code-conventions)
+  - [Git and GitHub](#git-and-github)
+  - [Python](#python)
+  - [Markdown](#markdown)
+- [Testing](#testing)
+- [Documentation](#documentation)
+
+## Code of Conduct
+
+Please read [`CODE_OF_CONDUCT.md`][code-of-conduct] before contributing.
+
+## Getting started
+
+To start contributing, first ensure your system meets the [requirements][readme-requirements]. Next, install the
+required Python packages, and [pre-commit hooks][pre-commit] using:
+
+```shell
+make requirements
+```
+
+It is better to use the above make command, rather than `pip install -r requirements.txt` and `pre-commit install`, as
+the command will ensure your pre-commit hooks are up-to-date with any changes made.
+
+The pre-commit hooks are a security feature to ensure no secrets, large data files, and Jupyter notebooks are
+accidentally committed into the repository. For more information about the pre-commit hooks used in this repository,
+see the [documentation][docs-pre-commit-hooks].
+
+## Code conventions
+
+We mainly follow [The GDS Way][gds-way] in our code conventions.
+
+### Git and GitHub
+
+We use Git to version control the source code; please read [The GDS Way][gds-way-git] for further details, including
+information about writing good commit messages, using `git rebase` for local branches, and `git merge --no-ff` for
+merges, as well as the entry on `git push --force-with-lease` instead of `git push -f`.
+
+If you want to modify the `.gitignore` files, see the template [documentation][docs-updating-gitignore] for further
+details.
+
+Our source code is stored on {{ cookiecutter.repository_hosting_platform }}. Pull requests into `master` require at
+least one approved review.
+
+### Python
+
+For Python code, we follow [The GDS Way Python style guide][gds-way-python] with a line length of 120; the flake8
+pre-commit hook should help with this!
+
+### Markdown
+
+Local links can be written as normal, but external links should be referenced at the bottom of the Markdown file for
+clarity. For example:
+
+```md
+Use a local link to reference the [`README.md`](./README.md) file, but an external link for [GOV.UK][gov-uk].
+
+[gov-uk]: https://www.gov.uk/
+```
+
+We also try and wrap Markdown to a line length of 120 characters, but this is not strictly enforced in all cases, for
+example with long hyperlinks.
```

Comment:
Is it worthwhile mentioning that you could turn off the pre-commit check for these hyperlinks by adding the following?
```
  # E501: noqa
```

Reply:
Ah I didn't think flake8 applied to Markdown files. Is this tripping the pre-commit hook on your side? I'd rather leave methods to circumvent the pre-commit hooks outside of this docs (users can look it up elsewhere!)